docs: Rename python configuration options and add deprecation alerts

mgaligniana opened the pull request:
Rename `request_bodies` to `max_request_body_size` and add deprecation alert.
Add deprecation alert also for `with_locals`.

<!-- Use this checklist to make sure your PR is ready for merge. You may delete any sections you don't need. -->

## Pre-merge checklist

- [ ] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

This PR adds the documentation for the changes made in https://github.com/getsentry/sentry-python/pull/2247 for ticket https://github.com/getsentry/sentry-python/issues/926.

A preview of how it looks like:

<img width="369" alt="image" src="https://github.com/getsentry/sentry-docs/assets/16822952/df054004-a2a9-41a0-a2c3-41dea2b63de8">

## Legal Boilerplate

<!-- Sentry employees and contractors can delete or ignore this section. -->

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

mgaligniana commented on src/platforms/common/configuration/options.mdx:
```diff
@@ -379,6 +396,12 @@ The number of characters after which the values containing text in the event pay
 
 <ConfigKey name="with-locals" supported={["python"]}>
```
I noticed this option was also deprecated: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/client.py#L82C20-L82C20. So I added the alert here too!

sentrivana reviewed (changes requested):
Thanks for adding the change to the docs! Looks good to me text-wise, what I think we can improve on a bit is the organization.

Right now the config key list is a mixture of deprecated and supported options. There's a `#### Deprecated` heading a bit further down the page, we should move the `request-bodies` and `with-locals` there. I'd also remove the description for those and just leave the deprecation note there with the link to the new config option.

mgaligniana commented:
I moved the two options to the bottom and added the "deprecated" header (I removed the text because I think is "repetitive", what do you think?) -> in the PR description I updated an image how it looks like now!

And I updated the diff too! I realized `python` was the last one to move inside of `supported=`, because the `max-request-body-size` description already exists.

Thank you!

sentrivana commented:
> I moved the two options to the bottom and added the "deprecated" header (I removed the text because I think is "repetitive", what do you think?) -> in the PR description I updated an image how it looks now!

Looks great!

> And I updated the diff too! I realized `python` was the last one to move inside of `supported=`, because the `max-request-body-size` description already exists.

Nicely spotted :) Yes, that makes sense.

Looks good to me, handing this over to our docs team.

sentrivana commented:
@mgaligniana Just noticed we reference `request_bodies` on [this page](https://github.com/getsentry/sentry-docs/edit/master/src/platforms/python/common/data-collected.mdx) as well, could you please update it to the new name there?

mgaligniana commented:
Ohh, good catch! Sure!
